src/server/rdb_save.cc

```diff
@@ -430,6 +432,7 @@ error_code RdbSerializer::SaveListObject(const PrimeValue& pv) {
         RETURN_ON_ERR(SaveLzfBlob(Bytes{reinterpret_cast<uint8_t*>(data), compress_len}, node->sz));
       } else {
         RETURN_ON_ERR(SaveString(node->entry, node->sz));
+        MaybeApplyFlushFunc();
```

Comment:
This is not complete but enough to reduce the memory footprint of the most use cases. We only serialize our own data structures (Which are the default) but not the redis one. Also `json` serializes to a string `blob` and it's rather difficult to make it work (unless we provide our own wrappero or something).